rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -1343,6 +1343,38 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|!REQUEST_COOKIES:/_pk_ref/|REQU
     setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/SQL_INJECTION-%{MATCHED_VAR_NAME}=%{tx.0}'"
 
 
+#
+# -=[ LibInjection Check on last path segment ]=-
+#
+# This is a sibling of rule 942100 that adds checking of the last path segment.
+#
+SecRule REQUEST_FILENAME "@rx ^/+(.*)$" \
```

Comment:
I don't think this do what you think it does. REQUEST_FILENAME will have the whole path, not the last segment.
Please update the comment.

Can you explain a bit more why you are using @rx and not simply passing the path to  @detectSQLi?

Reply:
Thanks for pointing that out. I don't think there is a variable then, which holds that information, is there? I'll update the regular expression to parse out the last segment then.

libinjection is more likely to fail when passing the path, at least that is what my tests showed. E.g. the following string produces a match `999999.1 union select unhex(hex(version())) -- and 1=1` while this doesn't `/999999.1 union select unhex(hex(version())) -- and 1=1` (tested directly with libinjection). So to be safe I don't want the slashes in there.

Maybe I could pass the entire path with the slashes replaced... But I don't know what that would do to the detection capabilities.

Reply:
You have https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual-%28v2.x%29#REQUEST_FILENAME.

Reply:
Sorry, I pasted the wrong link. I've updated it now.
Can you use that instead and squash the commits?